Keas.Mvc/ClientApp/components/Access/AccessAssignmentContainer.tsx

```diff
@@ -15,199 +16,112 @@ import { PermissionsUtil } from '../../util/permissions';
+      const accessFetchUrl = `/api/${context.team.slug}/access/listAssigned?personId=${props.person.id}`;
+      let accesses: IAccess[] = null;
+      try {
+        accesses = await context.fetch(accessFetchUrl);
+      } catch (err) {
+        toast.error('Error loading access list. Please refresh and try again.');
+      }
 
-    return (
-      <div>
-        {isRevokeModalShown && (
-          <RevokeAccess
-            assignment={this.state.selectedAssignment}
-            revoke={this.callRevoke}
-            cancelRevoke={this.hideModals}
-          />
-        )}
-        {isEditModalShown && (
-          <UpdateAccess
-            assignment={this.state.selectedAssignment}
-            update={this.callUpdate}
-            cancelUpdate={this.hideModals}
-          />
-        )}
-        {isAssignModalShown && (
-          <AssignAccess
-            closeModal={this.hideModals}
-            modal={isAssignModalShown}
-            person={this.props.person}
-            tags={this.context.tags}
-            selectedAccess={this.props.access}
-            onCreate={this.callAssign}
-          />
-        )}
-        <AccessAssignmentCard openAssignModal={this._openAssignModal}>
-          {this.props.person ? (
-            <AccessList
-              showDetails={this._openDetails}
-              personView={true}
-              access={this.state.assignments.map(
-                assignment => assignment.access
-              )}
-              onRevoke={access =>
-                this.showRevokeModal(
-                  assignments.find(
-                    assignment =>
-                      assignment.accessId === access.id &&
-                      assignment.personId === this.props.person.id
-                  )
-                )
-              }
-            />
-          ) : (
-            <AssignmentTable
-              assignments={assignments}
-              onRevoke={this.showRevokeModal}
-              onEdit={this.showEditModal}
-            />
-          )}
-        </AccessAssignmentCard>
-      </div>
-    );
-  }
+      return accesses.map(access => ({
+        ...access.assignments.find(
+          assignment =>
+            assignment.accessId === access.id &&
+            assignment.personId === props.person.id
+        ),
+        access
+      }));
+    };
 
-  private showRevokeModal = (assignment: IAccessAssignment) => {
-    this.setState({
-      selectedAssignment: assignment
-    });
-    this.props.history.push(
-      `${this._getBaseUrl()}/accessAssignment/revoke/${assignment.id}`
-    );
+    const getAssignments = async () => {
+      if (assignments.length === 0 && props.person) {
+        const assignmentsData = await fetchAssignments();
+        setAssignments(assignmentsData);
+        setSelectedAssignment(
+          assignmentsData.find(el => el.id === parseInt(params.id, 10))
+        );
+      }
+    };
+
+    getAssignments();
+  }, [
+    assignments.length,
```

Comment:
why only if the length changes but not the assignments themselves (or at least the reference to?)

Reply:
Before, eslint just complains that I need to include assignments.length, but I just changed it to assignments and it seems fine with that.